packages/web/lib/hooks/useUserXp.ts

```diff
@@ -0,0 +1,97 @@
+import { Constants } from '@metafam/utils';
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+import { SCAccount, SCAccountsData } from 'sourcecred';
+
+interface XPProps {
+  userTotalXP: number;
+  variationThisWeek: number;
+  variationLastWeek: number;
+  thisWeekXP: number;
+  lastWeekXP: number;
+  userWeeklyCred: number[];
+}
+
+export const useUserXP = (userAddress: string): XPProps => {
+  const [XPDataObj, setXPDataObj] = useState<XPProps>({
+    userTotalXP: 0,
+    variationThisWeek: 0,
+    variationLastWeek: 0,
+    thisWeekXP: 0,
+    lastWeekXP: 0,
+    userWeeklyCred: [],
+  });
+  useEffect(() => {
+    if (ethers.utils.isAddress(userAddress))
+      getXP(userAddress).then((res) => {
+        if (res) setXPDataObj(res);
+      });
+  }, [userAddress]);
+  return XPDataObj;
+};
+
+const getXP = async (userAddress: string): Promise<XPProps | null> => {
+  try {
+    const accountsData: SCAccountsData = await (
+      await fetch(Constants.SC_ACCOUNTS_FILE)
+    ).json();
+
+    const scAccount = await accountsData.accounts.find((account) =>
+      filterAccount(account, userAddress),
```

Comment:
This filtering is done by the find function using the account filter.

```suggestion
      accountFilter(account, userAddress),
```